zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/user/DbUserState.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.user;
+
+import static io.camunda.zeebe.util.buffer.BufferUtil.wrapString;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.engine.state.immutable.UserState;
+import io.camunda.zeebe.engine.state.mutable.MutableUserState;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+import io.camunda.zeebe.protocol.impl.record.value.user.UserRecord;
+import org.agrona.DirectBuffer;
+
+public class DbUserState implements UserState, MutableUserState {
+
+  private final UserRecordValue userRecordToRead = new UserRecordValue();
+  private final UserRecordValue userRecordToWrite = new UserRecordValue();
+
+  private final DbString username;
+  private final ColumnFamily<DbString, UserRecordValue> userColumnFamily;
```

Comment:
❓ How does multi-tenancy fit with user management? Is it possible to have multiple users with the same name in a different tenant?

Reply:
Good question, I can clarify this however my interpretation of multi-tenancy is that its the logical separation of data, and more specifically in our model data is deployed to a specific tenant and Users can then be assigned (given access) to those tenants. Based on that last part and as users are above tenants (in our implementation) its not possible for multiple users with the same name to exist